TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -31,25 +32,100 @@ private FullSelect CreateFullSelect(CodeElementsParser.FullselectContext context
             SubSelect subSelect = null;
             if (context.subselect() != null)
             {
-                subSelect = CreateSubSelect(context.subselect());
+                 subSelect = CreateSubSelect(context.subselect());
             }
-
             return new FullSelect(subSelect);
         }
-
         private SubSelect CreateSubSelect(CodeElementsParser.SubselectContext context)
         {
             FromClause fromClause = null;
             SelectClause selectClause = null;
             if (context.sql_selectClause() != null)
             {
                 selectClause = CreateSelectClause(context.sql_selectClause());
+                fromClause = CreateFromClause(context.sql_selectClause());
             }
-
             return new SubSelect(selectClause, fromClause);
+              
+        }
+
+        private FromClause CreateFromClause(CodeElementsParser.Sql_selectClauseContext context)
```

Comment:
Several remarks on this method:
- many contexts are used without checing prior if they're null or not. This code may work on correctly formed SQL code but will fail (throw exception) on incorrect syntax
- the use of `#region` is not a good practice. Instead you should create methods to factorize the code that you identify as working on a particular part/topic
- you create several SymbolReference here but we already have a CreateSymbolReference method which should be used as much as possible. If not applicable, see how you can factorize code